src/components/Carousel.js

```diff
@@ -97,6 +97,7 @@ class Carousel extends Component {
         }
 
         if (nextProps.autoPlay !== this.props.autoPlay) {
+            this.props.autoPlay = nextProps.autoPlay;
```

Comment:
You can't/shouldn't mutate props in react :)

Reply:
All right. Then how is it supposed to work? I mean when we later call `autoPlay`, that will still see the old `props.autoPlay` and do nothing. How else can that be addressed?

Reply:
moving `autoPlay` to `this.state` and mutate that would be ok?

Reply:
Yeh the problem here is you are already receiving new value, so to assign it early is likely an issue. Nextprops already has new value.

Issue is that this is WillReceiveProps, so nextprops is not yet the new props. Having an autoPlay state that is updated here with new value from props would work, and then everywhere else should use the state. Just need to check the initial initialisation in constructor handles copying the property to initial state

(Interestingly this copying of prop to state is exactly the way forward as they are starting deprecating the prop lifecycle methods in favour of getDerivedStateFromProps)

Reply:
I understand why it's bad to assign props (idk however why is it allowed at all by react). I add `autoPlay` to state then.